services/src/main/java/org/keycloak/organization/admin/resource/OrganizationMemberResource.java

```diff
@@ -79,32 +79,23 @@ public OrganizationMemberResource(KeycloakSession session, OrganizationModel org
 
     @POST
     @Consumes(MediaType.APPLICATION_JSON)
-    public Response addMember(UserRepresentation rep) {
+    public Response addMember(String id) {
         auth.realm().requireManageRealm();
-        if (rep == null || !Objects.equals(rep.getUsername(), rep.getEmail())) {
-            throw ErrorResponse.error("To add a member to the organization it is expected the username and the email is the same.", Status.BAD_REQUEST);
+        UserModel user = session.users().getUserById(realm, id);
```

Comment:
Isn't this handled by `UserStorageManager`?

Reply:
@martin-kanis please correct me if I am wrong ... I believe the question or concern might come from a fact that there is no control over those users who are not local keycloak users, so any update to those users (such as messing up with `kc.org` attribute) might be done outside of Keycloak. 

Would there be needed some mechanism to prevent/detect this kind of outside updates?

Reply:
For LDAP, we can do that as attributes like this one are very likely to be managed through a mapper. For custom providers, it will depend on how they are managing the user attributes.

Federated users (from custom providers) have their attributes stored in a specific table in Keycloak. I think we can handle such cases as well.

We can discuss this in a follow-up if you guys want as this PR is blocking others to go in.

Reply:
Yep, a mapper would probably be in place for cases like this if we are to propagate the persistence of the org attribute to LDAP. But the question still remains - if an admin changes the org in LDAP, how are we going to handle that? Will we need to validate these changes when re-importing/validating the user?

Reply:
If we don't allow creating a mapper to `kc.org` we don't need to worry about this, I would say.

The real problem, I would say, is with custom user providers because they have access to internals and can easily bypass user profile or any other validation we enforce elsewhere.